docs/technical/deployer_contract.md

```diff
@@ -93,69 +115,95 @@ status:
+#### 4. Deployer Logic and Status
+Now the deployer should do its magic. First it must set the status field `lastReconcileTime` on the current time thereby
+signaling the Landscaper that a deployer has picked up the deploy item.
 
-For a detailed description of the different timeouts, please have a look at [DeployItem Timeouts](../usage/DeployItemTimeouts.md).
+As long as the deployer is actually doing something - or waiting for something - `deployItemPhase` must be set on 
+`Processing` or `Deleting` and `jobId` remains different from `jobIdFinished`. 
 
-#### 4. Handle Generation
-Another indicator that something needs to be done is when `status.observedGeneration` differs from `metadata.generation`. The latter one changes every time the `spec` is modified and a difference in both shows that the deployer has not yet reacted on the latest changes to this deploy item. For this logic to work, the deployer has to set `status.observedGeneration` to the deploy item's generation at the beginning of the reconcile loop. Similarly to the reconcile annotation, the deployer should set the phase of the deploy item to `Init` if it updated the observed generation.
+Some deployers need to store information in the deploy item's status during or after processing it.
 
-> There is an auxiliary method `HandleAnnotationsAndGeneration` that handles steps 3 and 4 [defined here](../../pkg/deployer/lib/utils.go).
+A deploy item is deleted by Landscaper. The deployer see this at the deletion timestamp. In such a situation, the deployer
+should uninstall the artefacts from the target and if this was successfull remove the finalizers from the deploy item.
 
-#### 5. Check for Need for Action
-For most deployers, there probably isn't anything to do now if the deploy item is still in a final state (phase `Succeeded` or `Failed`) - it was finished before and nothing has changed, so the reconcile can be aborted at this point. Please note that this does not apply to all deployers and only works if the phase is actually set to `Init` when a reconcile annotation or an outdated observed generation is found.
+There is the following important annotation that needs to be handled by the deployer in case of a deletion:
+- `landscaper.gardener.cloud/delete-without-uninstall: true`: If this annotation is set at the deploy item and the deploy
+  item is deleted by the Landscaper, the deployer should remove the finalizer also if uninstalling the deployed artefacts
+  failed or even skip this uninstallation step at all.
```

Comment:
```suggestion
There is the following important annotation that needs to be handled by the deployer in case of a deletion:
- `landscaper.gardener.cloud/delete-without-uninstall: true`: If this annotation is set at the deploy item and the deploy
  item is deleted by the Landscaper, the deployer should remove the finalizer and skip the uninstallation step at all.
```